python/python/tests/test_async_client.py

```diff
@@ -1803,6 +1804,137 @@ async def test_zmscore(self, redis_client: TRedisClient):
         with pytest.raises(RequestError):
             await redis_client.zmscore(key2, ["one"])
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_zinterstore(self, redis_client: TRedisClient):
+        key1 = "{testKey}:1-" + get_random_string(10)
+        key2 = "{testKey}:2-" + get_random_string(10)
+        key3 = "{testKey}:3-" + get_random_string(10)
+        range = RangeByIndex(0, -1)
+        members_scores1 = {"one": 1.0, "two": 2.0}
+        members_scores2 = {"one": 1.5, "two": 2.5, "three": 3.5}
+
+        assert await redis_client.zadd(key1, members_scores=members_scores1) == 2
+        assert await redis_client.zadd(key2, members_scores=members_scores2) == 3
+
+        assert await redis_client.zinterstore(key3, [key1, key2]) == 2
+        assert await redis_client.zrange_withscores(key3, range) == {
+            "one": 2.5,
+            "two": 4.5,
+        }
+
+        # Intersection results are aggregated by the MAX score of elements
+        assert (
+            await redis_client.zinterstore(key3, [key1, key2], AggregationType.MAX) == 2
+        )
+        assert await redis_client.zrange_withscores(key3, range) == {
+            "one": 1.5,
+            "two": 2.5,
+        }
+
+        # Intersection results are aggregated by the MIN score of elements
+        assert (
+            await redis_client.zinterstore(key3, [key1, key2], AggregationType.MIN) == 2
+        )
+        assert await redis_client.zrange_withscores(key3, range) == {
+            "one": 1.0,
+            "two": 2.0,
+        }
+
+        # Intersection results are aggregated by the SUM score of elements
+        assert (
+            await redis_client.zinterstore(key3, [key1, key2], AggregationType.SUM) == 2
+        )
+        assert await redis_client.zrange_withscores(key3, range) == {
+            "one": 2.5,
+            "two": 4.5,
+        }
+
+        # Scores are multiplied by 2.0 for key1 and key2 during aggregation.
+        assert (
+            await redis_client.zinterstore(
+                key3, [(key1, 2), (key2, 2)], AggregationType.SUM
+            )
+            == 2
+        )
+        assert await redis_client.zrange_withscores(key3, range) == {
+            "one": 5.0,
+            "two": 9.0,
+        }
+
+        assert (
+            await redis_client.zinterstore(key3, [key1, "{testKey}-non_existing_key"])
+            == 0
+        )
+
```

Comment:
I'm adding tests for cross-slot scenario in another PR - just FYI

Reply:
Addressed.